lcdtest.py

```diff
@@ -0,0 +1,168 @@
+        print('   charmap - Charmap can be either A00 or A02. If your display contains Japanese')
+        print('              characters, it probably uses the A00 charmap, otherwise A02.')
+        print('              default=A00')
+        print('')
+        print('Options format: name=value e.g. expander=MCP23008 (Example below)')
+        print('%s gpio show_charmap cols=20 mode=GPIO.BCM rows=4 rs=15 rw=18 e=16 data=21,22,23,24 bl=None charmap=A00' % sys.argv[0])
+    else:
+        print('<options> For info about options run:')
+        print('')
+        print('   %s i2c' % sys.argv[0])
+        print('   %s gpio' % sys.argv[0])
+        print('')
+
+    sys.exit(1)
+
+
+if __name__ == '__main__':
+    if len(sys.argv) < 3:
+        print_usage()
+
+    lcdmode = sys.argv[1]
+    test = sys.argv[2]
+
+    # Parse options into a dictionary
+    options = {}
+    if sys.version_info > (3, 0):
+            options = dict([arg.split(sep='=', maxsplit=1) for arg in sys.argv[3:]])
+    else:
+        options = dict([arg.split('=', 1) for arg in sys.argv[3:]])
+
+    # Common options
+    cols = int(options.pop('cols'))
+    rows = int(options.pop('rows'))
+    charmap = options.pop('charmap', 'A00')
+    mode = GPIO.BCM if options.pop('mode', GPIO.BCM) == 'BCM' else GPIO.BOARD
+
+    if lcdmode == 'i2c':
+        if len(sys.argv) < 5:
+            print_usage()
+
+        # i2c options
+        i2c_expander = options.pop('expander')
+        address = int(options.pop('addr'), 16)
+        port = int(options.pop('port', '1'))
+
+        lcd = i2c.CharLCD(i2c_expander, address, cols=cols, port=port,
+                          rows=rows, expander_params=options)
+    elif lcdmode == 'gpio':
+        if len(sys.argv) < 8:
+            print_usage()
+        # gpio options
+        data = options.pop('data')
+        rs = int(options.pop('rs'))
+        e = int(options.pop('e'))
+        rw = int(options.pop('rw'))
+        bl = options.pop('bl', None)
+        if bl is not None:
+            bl = int(bl)
+        # Parse data pins into a list
+        pins_data = {}
+        if sys.version_info > (3, 0):
+            pins_data = data.split(sep=',')
+            print(pins_data)
+        else:
+            pins_data = data.split(',')
+        # Convert data pins to int
+        for i in range(len(pins_data)):
+            pins_data[i] = int(str(pins_data[i]))
+        print(pins_data)
+        lcd = gpio.CharLCD(pin_rs=rs, pin_rw=rw, pin_e=e, pins_data=pins_data,
+                       pin_backlight=bl, numbering_mode=mode, cols=cols, rows=rows,
+                       charmap=charmap)
+        pass
+    else:
+        print_usage()
+
+    # Run selected test
+    if test == 'show_charmap':
+        show_charmap.run(lcd, rows, cols)
+    elif test == 'format_test':
+        if ((cols == 20) and (rows == 4)):
+            format_test_20x4.run(lcd)
+        elif ((cols == 16) and (rows == 2)):
```

Comment:
You could drop those parentheses here and on line 161:

`if cols == 16 and rows == 2`